src/index.js

```diff
@@ -1,6 +1,6 @@
 'use strict'
 
-exports.DAGNode = require('./dag-node.js')
+exports.DAGNode = require('./dag-node-factory.js')
```

Comment:
We can force the users to always use the 'create', but exposing the Class does not make any instance mutable though.